src/components/drawing/index.js

```diff
@@ -492,6 +690,20 @@ drawing.singlePointStyle = function(d, sel, trace, fns, gd) {
 
             drawing.gradient(sel, gd, gradientID, gradientType,
                 [[0, gradientColor], [1, fillColor]], 'fill');
+        } else if(patternShape) {
+            var patternBGColor = getPatternAttr(markerPattern.bgcolor, d.i, null);
+            var patternSize = getPatternAttr(markerPattern.size, d.i, 8);
+            var patternSolidity = getPatternAttr(markerPattern.solidity, d.i, 0.3);
+            var perPointPattern = Array.isArray(markerPattern.shape) ||
+                                  Array.isArray(markerPattern.bgcolor) ||
+                                  Array.isArray(markerPattern.size) ||
+                                  Array.isArray(markerPattern.solidity);
```

Comment:
Shouldn't we use `Lib.isArrayOrTypedArray` here?

Reply:
Fixed in 51614783013bdc48ea9d8076f44a31a8da95b5a5